driver-core/src/test/unit/com/mongodb/internal/connection/AbstractConnectionPoolTest.java

```diff
@@ -359,9 +457,89 @@ protected DefaultConnectionPool getPool() {
         List<Object[]> data = new ArrayList<Object[]>();
         for (File file : JsonPoweredTestHelper.getTestFiles("/connection-monitoring-and-pooling")) {
             BsonDocument testDocument = JsonPoweredTestHelper.getTestDocument(file);
-            data.add(new Object[]{file.getName(), testDocument.getString("description").getValue(), testDocument});
+            data.add(new Object[]{
+                    file.getName(),
+                    testDocument.getString("description").getValue(),
+                    testDocument,
+                    JsonTestServerVersionChecker.skipTest(testDocument, BsonDocument.parse("{}"))
+            });
         }
         return data;
     }
 
+    private enum Style {
+        UNIT,
+        INTEGRATION;
+
+        public static Style of(final String name) {
+            return valueOf(name.toUpperCase());
+        }
+    }
+
+    private static final class ConnectionIdAdjustingConnectionPool implements ConnectionPool {
+        private static final int UNINITIALIZED = Integer.MAX_VALUE;
+
+        private final DefaultConnectionPool pool;
+        private final AtomicInteger connectionIdLocalValueAdjustment;
+
+        private ConnectionIdAdjustingConnectionPool(final DefaultConnectionPool pool) {
```

Comment:
I was not thinking straight and confused myself during our call by saying that I need to invalidate the `pool` here to prevent `ConnectionIdAdjustingConnectionPool` from observing connections that were created previously. This is not needed because the pool passed here is always a new pool.